core/charm/repository/charmhub.go

```diff
@@ -479,8 +562,15 @@ func (c *CharmHubRepository) listResourcesMap(id corecharm.CharmID) (map[string]
 	if resp.Error != nil {
 		return nil, errors.Annotatef(errors.New(resp.Error.Message), "listing resources for charm %q", curl.String())
 	}
-	results := make(map[string]charmresource.Resource, len(resp.Entity.Resources))
-	for _, v := range resp.Entity.Resources {
+	return resp.Entity.Resources, nil
+}
+
+func transformResourceRevision(resources []transport.ResourceRevision) (map[string]charmresource.Resource, error) {
```

Comment:
Is this just for making a charmresource.Resource slice out of these ResourceRevisions ? A small comment might be helpful here